openfeature/event_executor.go

```diff
@@ -215,6 +213,14 @@ func (e *eventExecutor) emitOnRegistration(
 	}
 }
 
+func (e *eventExecutor) State(domain string) State {
```

Comment:
Question: What is the convention for casing of internal interfaces? This implements `clientEvent`, which is internal, but it's methods are capitalized. Is that convention? Since it's internal they could all be lower case. :thinking:

Reply:
I don’t think there’s a clear convention here. Ultimately, it comes down to the simple question of whether you want others to implement this interface. To me, this (and similar interfaces) seems like internal code, so it makes sense for the methods to be in lowercase—I agree with that.

The issue is that the methods already include exported ones. (Keep in mind that it doesn’t matter if the interface itself is exported; what matters is whether the methods are.) Changing this now would introduce a breaking change.

Reply:
That makes sense!